tests/thread_commission_test.go

```diff
@@ -0,0 +1,78 @@
+package tests
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/canonical/matter-snap-testing/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	otbrSnap = "openthread-border-router"
+
+	defaultInfraInterfaceValue = "wlan0"
+	infraInterfaceKey          = "infra-if"
+	localInfraInterfaceEnv     = "LOCAL_INFRA_IF"
+)
+
+func TestAllClustersAppThread(t *testing.T) {
+	// Start clean
+	utils.SnapRemove(t, otbrSnap)
+
+	t.Cleanup(func() {
+		cleanupRemoteDevice(t)
+	})
+
+	// Remote device: Access remote device, setup Thread network, and get active dataset
+	activeDataset := setupRemoteThreadNetwork(t)
+	trimmedActiveDataset := strings.TrimSpace(activeDataset)
```

Comment:
The active dataset needed to commission should be from a network formed on the local device. The OTBR agent on the remote device is just enabling communication between the matter app and RCP.

Reply:
Initial design:
```
Local Machine:
- chip-tool snap
- OTBR snap with otbr-agent
- all-clusters snap

Remote Machine:
- OTBR snap with Thread network
```
current design:
```
Local Machine:
- chip-tool snap
- OTBR snap with Thread network

Remote Machine:
- OTBR snap with otbr-agent
- all-clusters snap
```